src/borg/archiver/extract_cmd.py

```diff
@@ -56,32 +56,53 @@ def do_extract(self, args, repository, manifest, archive):
         else:
             pi = None
 
-        for item in archive.iter_items(filter, preload=True):
+        for item in archive.iter_items(preload=True):
```

Comment:
Hmm, this develops to be a bit more tricky than I initially thought it is.

If you leave away the filter here (which at first looks like a good idea to get ALL items yielded), it will preload the file content chunks of ALL yielded items (not just the ones we might extract).

That's not necessary and it is even harmful, leading to a memory leak if one does not retrieve these chunks, see comment in `unpack_many`.

Reply:
When you use the filter, only items that match the filter criteria are processed which prevents us from logging any excluded items, which is why I proposed the "two pass" solution but that would display excludes either at the beginning or the end.

Reply:
Sure. So there is no good solution yet and it is not easy due to the preloading requirements.

Reply:
I am working right now on decoupling preloading and item iteration to help you with that.

After that is merged, you can rebase your work on these changes.

Reply:
Thank you so much. I really appreciate it!

Reply:
#8592 was just merged into master.

Please update your local master branch and rebase your PR branch onto that.

As you can see there, `archive.iter_items()` does not support preloading anymore.
At the places where preloading is needed, I inserted a new call that does the preloading.

Be careful: the preload call must only be done if the item chunks will be fetched later (e.g. if the item is extracted, because it is included / not excluded). The `borg extract` code fetches the content chunks even if it is in dry-run mode, because it is documented to do all steps of the extraction except writing content to disk files (so people can use it as a practical test whether the extraction would work without needing the space to do the extraction).

Reply:
Thank you so much once again for the clarification and your help! Working on it.